pinot-core/src/main/java/org/apache/pinot/core/query/aggregation/function/DistinctAggregationFunction.java

```diff
@@ -139,6 +140,12 @@ public void aggregate(int length, AggregationResultHolder aggregationResultHolde
 
   @Override
   public DistinctTable extractAggregationResult(AggregationResultHolder aggregationResultHolder) {
+    if (_distinctTable == null) {
+      ColumnDataType[] columnDataTypes = new ColumnDataType[_columnNames.length];
+      Arrays.fill(columnDataTypes, ColumnDataType.STRING);
```

Comment:
why do we set them as String. we should use the type based on the response from the underlying block rt. Even if the block is empty, I think the dataTypes must be set properly.

Reply:
+1

Reply:
Cant think of a better solution other than this for now. Let's go with it. Make sure you check for size of the distinct table in the merge as well.